spec/MongoStorageAdapter.spec.js

```diff
@@ -350,6 +350,135 @@ describe_only_db('mongo')('MongoStorageAdapter', () => {
     expect(postIndexPlan.executionStats.executionStages.stage).toBe('FETCH');
   });
 
+  it('should never use case_insensitive_email index for regex query on email', async () => {
+    const user = new Parse.User();
+    user.set('username', 'Bugs');
+    user.set('password', 'Bunny');
+    user.set('email', 'bugs@wabbits.com');
+    await user.signUp();
+
+    const database = Config.get(Parse.applicationId).database;
+    const schema = await new Parse.Schema('_User').get();
+
+    await database.adapter.ensureIndex(
+      '_User',
+      schema,
+      ['email'],
+      'case_insensitive_email',
+      true
+    );
+
+    const preIndexPlan = await database.find(
+      '_User',
+      { email: { $regex: 'bugs@wabbits.com' } },
+      { explain: true }
+    );
+
+    const preIndexNoRegex = await database.find(
+      '_User',
+      { email: { $eq: 'bugs@wabbits.com' } },
+      { explain: true, caseInsensitive: true }
+    );
+
+    await database.adapter.ensureIndex(
+      '_User',
+      schema,
+      ['email'],
+      'email_1',
+      false
+    );
+
+    const postIndexPlan = await database.find(
+      '_User',
+      { email: { $regex: 'bugs@wabbits.com' } },
+      { explain: true }
+    );
+
+    const postIndexNoRegex = await database.find(
+      '_User',
+      { email: { $eq: 'bugs@wabbits.com' } },
+      { explain: true, caseInsensitive: true }
+    );
+
+    await database.adapter.dropIndex('_User', 'email_1');
```

Comment:
I don't know why this is necessary but it appears that later tests error if another non-collated index on `email` or `username` exists already. I was unable to figure out where those other indexes are being created, but this definitely fixes the issue.

Reply:
That's a good question... I didn't remove that index because the only other test in the entire spec folder that adds an index, `should use index for caseInsensitive query`, does not remove its `case_insensitive_username` index. Like I said, I have no idea why I need to remove `email_1`, but it conflicted with other index creation later on so I did it. The other index didn't conflict with anything so I left it.

Reply:
Let's add a comment to convey why the index is being dropped.  Someone else might know more and have a better solution or it could highlight an issue that could / should be avoided by fixing the place where they are conflicting.